lib/service/servicecfg/proxy.go

```diff
@@ -148,7 +148,9 @@ func (c ProxyConfig) getDefaultAddr(port int) string {
 	host := "<proxyhost>"
 	// Try to guess the hostname from the HTTP public_addr.
 	if len(c.PublicAddrs) > 0 {
-		host = c.PublicAddrs[0].Host()
+		publicAddr := c.PublicAddrs[0]
+		host = publicAddr.Host()
+		port = publicAddr.Port(port)
```

Comment:
Was the port always the default before the fix? Will this suddenly change behavior in installed clusters, "fixing" ports that we are binding to default instead?

Reply:
OK, seems safe enough to push.